wgpu_macros/src/generic_bindings.rs

```diff
@@ -1023,7 +1023,7 @@ pub fn sub_module_generic_bindings(input: TokenStream) -> TokenStream {
         }
 
         impl <'a,  T : pipeline :: RuntimePass<'a>> #context<'a, T, #(#init),*> {
-            fn new() -> Self {
+            unsafe fn new() -> Self {
```

Comment:
Either way, I think this is the more idiomatic way to say that you shouldn't do something and that it's undefined behavior. Now even if someone does manage to, against all odds, create their own new context, they'll have to accept the blame.

Reply:
I have thought about making some of my methods unsafe. Like maybe writing to a buffer should also be unsafe. 

Does this change compile?

Reply:
Nvm, I looked at your full changes and it looks like you added the unsafe block